File: LCX/ValidEndPoints/dsl/examples/market_data_dynamic/market_data_dynamic.php
<?php
error_reporting(E_ALL);
ini_set('display_errors', 1);


function fetch_all_pairs_data(pair) {
    // GET /api/ticker
    $url = "https://exchange-api.lcx.com/api/ticker";
    $headers = ["Content-Type: application/json"];

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    echo json_encode(json_decode($response), JSON_PRETTY_PRINT);
    return json_decode($response, true);

}

# Workflow: FetchAllPairsData (sequence)

function get_ethusdc_ticker(pair) {
    // GET /api/ticker
    $url = "https://exchange-api.lcx.com/api/ticker";
    $headers = ["Content-Type: application/json"];

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    echo json_encode(json_decode($response), JSON_PRETTY_PRINT);
    return json_decode($response, true);

}

function get_lcxusdc_ticker(pair) {
    // GET /api/ticker
    $url = "https://exchange-api.lcx.com/api/ticker";
    $headers = ["Content-Type: application/json"];

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    echo json_encode(json_decode($response), JSON_PRETTY_PRINT);
    return json_decode($response, true);

}

File: LCX/ValidEndPoints/dsl/examples/market_data_dynamic/market_data_dynamic.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json;
using System.Security.Cryptography;
using System.Text;


public static async Task<dynamic> fetch_all_pairs_data(pair)
{
    // GET /api/ticker
    using (var client = new HttpClient())
    {
        var url = "https://exchange-api.lcx.com/api/ticker";
        client.DefaultRequestHeaders.Add("Content-Type", "application/json");

        try
        {
            var response = await client.GetAsync(url);
            response.EnsureSuccessStatusCode();
            var content = await response.Content.ReadAsStringAsync();
            Console.WriteLine(content);
            return JsonConvert.DeserializeObject(content);

        }
        catch (HttpRequestException e)
        {
            Console.WriteLine("Error: " + e.Message);
            throw;
        }
    }
}

# Workflow: FetchAllPairsData (sequence)

public static async Task<dynamic> get_ethusdc_ticker(pair)
{
    // GET /api/ticker
    using (var client = new HttpClient())
    {
        var url = "https://exchange-api.lcx.com/api/ticker";
        client.DefaultRequestHeaders.Add("Content-Type", "application/json");

        try
        {
            var response = await client.GetAsync(url);
            response.EnsureSuccessStatusCode();
            var content = await response.Content.ReadAsStringAsync();
            Console.WriteLine(content);
            return JsonConvert.DeserializeObject(content);

        }
        catch (HttpRequestException e)
        {
            Console.WriteLine("Error: " + e.Message);
            throw;
        }
    }
}

public static async Task<dynamic> get_lcxusdc_ticker(pair)
{
    // GET /api/ticker
    using (var client = new HttpClient())
    {
        var url = "https://exchange-api.lcx.com/api/ticker";
        client.DefaultRequestHeaders.Add("Content-Type", "application/json");

        try
        {
            var response = await client.GetAsync(url);
            response.EnsureSuccessStatusCode();
            var content = await response.Content.ReadAsStringAsync();
            Console.WriteLine(content);
            return JsonConvert.DeserializeObject(content);

        }
        catch (HttpRequestException e)
        {
            Console.WriteLine("Error: " + e.Message);
            throw;
        }
    }
}

